homeassistant/components/goalzero/config_flow.py

```diff
@@ -116,3 +124,26 @@ async def _async_try_connect(self, host):
             _LOGGER.exception("Unexpected exception")
             return None, "unknown"
         return str(api.sysdata["macAddress"]), None
+
+
+class GoalZeroOptionsFlowHandler(OptionsFlow):
+    """Handle Goal Zero options."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input: ConfigType | None = None):
+        """Manage Goal Zero options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        options = {
+            vol.Optional(
+                CONF_SCAN_INTERVAL,
+                default=self.config_entry.options.get(
+                    CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                ),
```

Comment:
In general: We don't want to expose a scan interval, unless absolutely not other solution is available.

Please elaborate on why this change is needed and why there isn't another solution available (and which the ones are you have explored).

Reply:
Push updates are not currently available in the api. The connected anywhere feature uses mqtt over tls to aws which is not currently something I can penetrate. The direct connect option only works with the smart phone app. If I could do the direct connect option, HA would have to directly connect to the device's access point which is not possible, right? 

This is a battery operated device. Every poll would obviously drain it a little. With this option, the user would be able to choose an interval more appropriate for their use case. If they keep it plugged in, they can afford to shorten it. If they plan to use it outside unplugged with solar (on a cloudy day), they may want to poll it less frequently.

Reply:
So the question is: Why not adjust the current polling rate to a sane default?

Alternatively, an option can be implemented to disable automatic updates completely.

Reply:
So I would directly replace scan interval with a disable automatic updates option or should this be in a service?

Reply:
> 30 seconds seems sane to me for a battery operated device, don't you think? 

I can't judge that, as I'm not a user of this integration/device.

>  So I would directly replace scan interval with a disable automatic updates option or should this be in a service?

Yes, there is now a PR open for waze/google travel time, implementing something similar.
Add an option to disable updates, user can update entities using the `homeassistant.update_entity` service on their own schedule and terms.

Reply:
PR: https://github.com/home-assistant/core/pull/51291 is now closed, will wait until the system options work is done.